lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -410,6 +410,50 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
         }
         return failure();
       });
+  patterns.onOp("LayerNormalization", 17,
+                [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+                  Torch::ValueTensorType Y_type;
+                  Torch::ValueTensorType Mean_type;
+                  Torch::ValueTensorType InvStdDev_type;
+                  Value X;
+                  Value Scale;
+                  Value B;
+                  int64_t axis;
+		  float epsilon;
+                  int64_t stash_type;
+                  if (binder.tensorOperandAtIndex(X, 0) ||
+                      binder.tensorOperandAtIndex(Scale, 1) ||
+                      binder.tensorOperandAtIndex(B, 2) ||
+                      binder.tensorResultTypeAtIndex(Y_type, 0)) 
+                    return failure();
+                  if (binder.tensorResultTypeAtIndex(Mean_type, 1) ||
+                      binder.tensorResultTypeAtIndex(InvStdDev_type, 2))
+                    return failure();
+                  if (binder.s64IntegerAttr(axis, "axis", -1) ||
+                      binder.f32FloatAttr(epsilon, "epsilon", 0.00001) ||
+                      binder.s64IntegerAttr(stash_type, "stash_type", 1)) 
+                    return failure(); 
+                  Value constEpsilon = rewriter.create<Torch::ConstantFloatOp>(
+                    binder.getLoc(), rewriter.getType<Torch::FloatType>(),
+                    rewriter.getF64FloatAttr(epsilon));
+                  unsigned rank = 1; 
+                  if(std::optional<unsigned> maybeRank = Torch::getTensorRank(X))
+                    rank = *maybeRank;
+                  SmallVector<Value> normalized;
+                  if(axis < 0)
+                    axis = axis + rank;
+                  for (int64_t n = axis; n < rank ; n++) {
+                    normalized.push_back(rewriter.create<Torch::ConstantIntOp>(
+                    binder.getLoc(), rewriter.getI64IntegerAttr(n)));
+                  }
+                  Value normalized_shape = rewriter.create<Torch::PrimListConstructOp>(
+                    binder.getLoc(),
+                    Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+                    normalized);
```

Comment:
This logic for computing the `normalizedShape` seems incorrect.

Reply:
Can you please tell us how did you infer this?

Reply:
See links above. The normlaizedShape is derived from axix. In onnx  you have axis and rank whereas in torch there is normalized shape. Please see the description in 
https://onnx.ai/onnx/operators/onnx__LayerNormalization.html

Reply:
Although the downstream pipeline does not make use of normalized shape values except the list size still based on this https://onnx.ai/onnx/operators/onnx__LayerNormalization.html#:~:text=If%20X%E2%80%99s%20shape%20is%20%5Bd%5B0%5D%2C%20...%2C%20d%5Baxis%2D1%5D%2C%20d%5Baxis%5D%2C%20...%2C%20d%5Brank%2D1%5D%5D%2C%20the%20shape%20of%20Mean%20and%20InvStdDev%20is%20%5Bd%5B0%5D%2C%20...%2C%20d%5Baxis%2D1%5D%2C%201%2C%20...%2C%201%5D.%20Y%20and%20X%20have%20the%20same%20shape.

The normalized shape list should contain all 1s.

Reply:
I resolved this issue please check the code. Yet I am understanding it in the way that normalized_shape contains dimensions from the input dimension. So if input is X[4,10,20,30] and axis is provided as maybe 2 then normalized_shape should be something like [20,30].